integration/npm_test.go

```diff
@@ -140,5 +143,89 @@ func testNPM(t *testing.T, context spec.G, it spec.S) {
 				Expect(env.NpmConfigLoglevel).To(Equal("error"))
 			})
 		})
+
+		context("when using CA certificates", func() {
+			var (
+				client *http.Client
+			)
+
+			it.Before(func() {
+				var err error
+				source, err = occam.Source(filepath.Join("testdata", "npm_ca_certs"))
+				Expect(err).NotTo(HaveOccurred())
+
+				caCert, err := ioutil.ReadFile(fmt.Sprintf("%s/client/ca.pem", source))
+				Expect(err).ToNot(HaveOccurred())
+
+				caCertPool := x509.NewCertPool()
+				caCertPool.AppendCertsFromPEM(caCert)
+
+				cert, err := tls.LoadX509KeyPair(fmt.Sprintf("%s/client/cert.pem", source), fmt.Sprintf("%s/client/key.pem", source))
+				Expect(err).ToNot(HaveOccurred())
+
+				client = &http.Client{
+					Transport: &http.Transport{
+						TLSClientConfig: &tls.Config{
+							RootCAs:      caCertPool,
+							Certificates: []tls.Certificate{cert},
+							MinVersion:   tls.VersionTLS12,
+						},
+					},
+				}
+			})
+
+			it("builds a working OCI image and uses a client-side CA cert for requests", func() {
+				var err error
+				var logs fmt.Stringer
+				image, logs, err = pack.WithNoColor().Build.
+					WithBuildpacks(nodeBuildpack).
+					WithPullPolicy("never").
+					Execute(name, filepath.Join(source, "server"))
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(logs).To(ContainLines(ContainSubstring("CA Certificates Buildpack")))
+				Expect(logs).To(ContainLines(ContainSubstring("Node Engine Buildpack")))
+				Expect(logs).To(ContainLines(ContainSubstring("NPM Install Buildpack")))
+				Expect(logs).To(ContainLines(ContainSubstring("NPM Start Buildpack")))
+
+				container, err = docker.Container.Run.
+					WithPublish("8080").
+					WithEnv(map[string]string{
+						"PORT":                 "8080",
+						"SERVICE_BINDING_ROOT": "/bindings",
+						"NODE_OPTIONS":         "--use-openssl-ca",
+					}).
+					WithVolume(fmt.Sprintf("%s/server/binding:/bindings/ca-certificates", source)).
+					Execute(image.ID)
+				Expect(err).NotTo(HaveOccurred())
+
+				Eventually(func() string {
+					cLogs, err := docker.Container.Logs.Execute(container.ID)
+					Expect(err).NotTo(HaveOccurred())
+					return cLogs.String()
+				}).Should(
+					ContainSubstring("Added 1 additional CA certificate(s) to system truststore"),
+				)
+
+				// give the app 1 second to get set up before we curl it
+				time.Sleep(1 * time.Second)
+
+				request, err := http.NewRequest("GET", fmt.Sprintf("http://localhost:%s/env", container.HostPort("8080")), nil)
```

Comment:
Shouldn't the server handle tls and this be an http**s** request?

Reply:
When I used `https`, it throws the following error:
```
        Unexpected error:
            <*url.Error | 0xc0002de480>: {
                Op: "Get",
                URL: "https://localhost:32855/env",
                Err: {
                    s: "http: server gave HTTP response to HTTPS client",
                },
            }
            Get "https://localhost:32855/env": http: server gave HTTP response to HTTPS client
        occurred
```

Reply:
I'm not sure why

Reply:
The NPM and Yarn apps I was using were setting an http server up instead of https. Will fix.

Reply:
Fixed in https://github.com/paketo-buildpacks/nodejs/pull/341/commits/df3be442dc9ce155db065af8d3010a2b16b8da24